File: services/Contracts/Core/Domain/Entities/Contract.cs
using System;
using Contracts.Core.Domain.Abstractions.Entities;
using Contracts.Core.Domain.Enums;

namespace Contracts.Core.Domain.Entities;

public class Contract : Entity
{
    public required Guid ConsumerId { get; set; }
    public required Guid ProviderId { get; set; }
    public required ContractStatus Status { get; set; }
}

File: services/ControlPanel/models/Contract.ts
import type { Entity } from "./Entity";
import type { Consumer } from "./Consumer";
import type { Provider } from "./Provider";

export type Contract = Entity & {

    status: ContractStatus;
    consumerId: string;
    providerId: string;
    consumer: Consumer;
    provider: Provider;
}

export enum ContractStatus {
    NotDefined = 1,
    Pending = 2,
    Approved = 2
}

export const ContractStatusOptions = [
    { value: ContractStatus.NotDefined, label: "Μη Ορισμένη" },
    { value: ContractStatus.Pending, label: "Εκκρεμή" },
    { value: ContractStatus.Approved, label: "Εγκεκριμένη" }
];
